front/app/modules/commercial/insights/services/insightsCategories.ts

```diff
@@ -72,14 +73,19 @@ export function updateInsightsCategory(
   );
 }
 
-export function deleteInsightsCategories(
-  insightsViewId: string,
-  insightsCategoryId: string
-) {
-  return streams.delete(
-    `${API_PATH}/${getInsightsCategoriesEndpoint(
-      insightsViewId
-    )}/${insightsCategoryId}`,
-    insightsCategoryId
+export async function deleteInsightsCategories(insightsViewId: string) {
+  const response = await streams.delete(
+    `${API_PATH}/${getInsightsCategoriesEndpoint(insightsViewId)}`,
+    ''
```

Comment:
I don't understand the streams enough to be sure it's needed, but elsewhere we add the 'data id', the insightsViewId in this case.. Perhaps @DavidCnoops can tell us if it's needed here.

Reply:
Hi, normally you are right! However, this is a bulk action (it deletes all categories, not a specific one), so in this case, it is not needed. :)

Reply:
dataId should allow for `null` I guess. :)

Reply:
As long as the deleted id's aren't being used in any other stream it should indeed be fine to leave out the id parameter. It would however be careful with this, as it migth not always be known if a deleted id is present in any other streams (or will be present in any other streams somewhere in the future), so as a best practice I'd advise to include the id parameter whenever possible. that's also the reason why today it's a required parameter (you can however make a small change to the steams delete function to make it optional).

Reply:
I also think in this case it's fine. Just wanted to provide a bit of context to caution people into overusing it in general :).